src/modules/dndImport/components/DndImportPanel.js

```diff
@@ -110,25 +110,29 @@ export class DndImportPanel extends MvuElement {
 	_importFile(dataTransfer) {
 		const translate = (key) => this._translationService.translate(key);
 		const files = dataTransfer.files;
-		const read = async (file) => {
-			const textContent = await file.text();
-			return textContent;
+
+		const importData = async (blob, sourceType) => {
+			const text = await blob.text();
+			setData(text, sourceType);
 		};
 		const handleFiles = (files) => {
-			Array.from(files).forEach(async f => {
+
+			Array.from(files).forEach(f => {
 				try {
-					const mimeType = f.type;
-					if (DragAndDropSupportedMimeTypes.includes(mimeType)) {
-						const data = await read(f);
-						setImportData(data, mimeType);
+					const sourceTypeResult = this._sourceTypeService.forBlob(f);
+					switch (sourceTypeResult.status) {
+						case SourceTypeResultStatus.OK:
+							importData(f, sourceTypeResult.sourceType);
+							break;
+						case SourceTypeResultStatus.MAX_SIZE_EXCEEDED:
+							emitNotification(translate('dndImport_import_max_size_exceeded'), LevelTypes.WARN);
+							break;
+						case SourceTypeResultStatus.UNSUPPORTED_TYPE:
+							emitNotification(translate('dndImport_import_unsupported'), LevelTypes.WARN);
+							break;
+						case SourceTypeResultStatus.OTHER:
+							emitNotification(translate('dndImport_import_unknown'), LevelTypes.ERROR);
```

Comment:
just as a suggestion: can we combine sourceTypeResult.status handling within one block of code for all SourceTypeService calls?

Reply:
Thx